llvm/lib/IR/DataLayout.cpp

```diff
@@ -193,9 +193,11 @@ constexpr DataLayout::PrimitiveSpec DefaultVectorSpecs[] = {
 };
 
 // Default pointer type specifications.
-constexpr DataLayout::PointerSpec DefaultPointerSpecs[] = {
+const DataLayout::PointerSpec DefaultPointerSpecs[] = {
```

Comment:
We could avoid the need for a global ctor/dtor if we just stored a plain integer value? Since we only support 0/-1 we could just rely on sign-extension? But of course that makes the getter slower so probably best to ignore this suggestion.

Reply:
I thought about that, but we might want to allow a target to specify that the `nullptr` is an arbitrary bit pattern to be more future proof such that LLVM will not try any folding regarding those casts involving `nullptr`. I added `'c'` as the flag to specify that.